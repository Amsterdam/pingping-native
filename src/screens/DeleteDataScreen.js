import React, {useState} from 'react';

import {useMutation} from '@apollo/client';
import AsyncStorage from '@react-native-community/async-storage';
import PropTypes from 'prop-types';
import {StyleSheet, Text, View, ScrollView} from 'react-native';

import {testIDs} from '../../e2e/modulesTestIDs';
import {resetStore} from '../apollo/apolloClient';
import DELETE_USER_MUTATION from '../apollo/Mutation/deleteUserMutation';
import TrashIcon from '../assets/svg/icons/TrashIcon';
import FilledHeader from '../components/header/FilledHeader';
import ContentLayout from '../components/layout/ContentLayout';
import DeleteDataModal from '../components/modals/DeleteDataModal';
import Container from '../components/shared/Container';
import Button from '../components/shared/RoundedButton';
import Body from '../components/typography/Body';
import Title from '../components/typography/Title';
import theme from '../config/theme';
import sentryHelper from '../helpers/sentryHelper';

const DeleteDataScreen = ({navigation, setLogOut}) => {
	const [open, setOpen] = useState(false);
	const [loading, setLoading] = useState(false);
	const [deleteUser] = useMutation(DELETE_USER_MUTATION);

	const doDeleteUser = async () => {
		setLoading(true);
		try {
			await deleteUser({
				variables: {
					confirm: 'delete',
				},
			});
			await AsyncStorage.clear();
			setLogOut();
			resetStore();
		} catch (error) {
			setLoading(false);
			sentryHelper(error.message);
		}
	};

	return (
		<Container
			testID={testIDs.DELETE_DATA.SCREEN}
			statusBarColor={theme.colors.headerColor}>
			<FilledHeader navigation={navigation} title="Privacy" />
			<ScrollView contentContainerStyle={styles.scrollView}>
				<ContentLayout>
					<Title style={styles.margin}>Jouw eigen gegevens</Title>
					<View style={styles.emojiContainer}>
						<Text>ðŸ‘†ðŸ˜Œ</Text>
					</View>
					<Body variant="b3" style={styles.margin}>
						Wij van Ping Ping gebruiken de door jouw ingevulde gegevens alleen
						om jouw route te bepalen en de app te verbeteren. Wij zullen nooit
						jouw gegevens verkopen aan andere partijen.
					</Body>
					<Body variant="b3" style={styles.margin}>
						Wil je toch graag je inloggegevens verwijderen? Druk simpelweg op de
						knop hieronder om je gegevens uit ons systeem te halen. Hierdoor
						gaat je voortgang verloren.
					</Body>
					<View style={styles.inputContainer}>
						<Button
							deleteButton
							icon={<TrashIcon height={20} color={theme.colors.white} />}
							label="Verwijder mijn gegevens"
							onPress={() => setOpen(true)}
							testid={testIDs.DELETE_DATA.DELETE_BUTTON}
							full
						/>
					</View>
				</ContentLayout>
			</ScrollView>
			<DeleteDataModal
				open={open}
				setOpen={setOpen}
				setLogOut={setLogOut}
				navigation={navigation}
				doDeleteUser={doDeleteUser}
				loading={loading}
			/>
		</Container>
	);
};

const styles = StyleSheet.create({
	margin: {
		marginBottom: 30,
	},
	scrollView: {
		backgroundColor: theme.colors.white,
		flex: 1,
	},
	inputContainer: {alignItems: 'center', justifyContent: 'center'},
	emojiContainer: {
		alignItems: 'center',
		marginBottom: 10,
	},
});

DeleteDataScreen.propTypes = {
	navigation: PropTypes.object.isRequired,
	setLogOut: PropTypes.func.isRequired,
};

export default DeleteDataScreen;
