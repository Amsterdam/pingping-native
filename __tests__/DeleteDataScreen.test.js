import React from 'react';

import { MockedProvider } from '@apollo/client/testing';
import renderer from 'react-test-renderer';

import DeleteDataScreen from '../src/screens/DeleteDataScreen';

jest.useFakeTimers();
const navigation = { navigate: jest.fn() };

test('renders correctly', () => {
	const tree = renderer
		.create(
			<MockedProvider mocks={[]}>
				<DeleteDataScreen navigation={navigation} />
			</MockedProvider>
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
