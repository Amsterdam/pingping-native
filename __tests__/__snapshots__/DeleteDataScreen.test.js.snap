// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
      },
      undefined,
    ]
  }
  testID="DELETE_DATA_SCREEN"
>
  <RCTSafeAreaView
    style={
      Array [
        Object {
          "flex": 0,
        },
        Object {
          "backgroundColor": "#0d2036",
        },
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#0d2036",
        "flexDirection": "row",
        "justifyContent": "center",
        "minHeight": 56,
        "position": "relative",
      }
    }
  >
    <View
      style={
        Object {
          "left": 5,
          "position": "absolute",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="HEADER_BACK_BUTTON"
      >
        <RNSVGSvgView
          bbHeight={24}
          bbWidth={24}
          fill="none"
          focusable={false}
          height={24}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 24,
                "width": 24,
              },
            ]
          }
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <RNSVGGroup
            fill={null}
            propList={
              Array [
                "fill",
              ]
            }
          >
            <RNSVGPath
              clipRule={0}
              d="M19.7 11H7.5l5.6-5.6L11.7 4l-8 8 8 8 1.4-1.4L7.5 13h12.2v-2Z"
              fill={4294967295}
              fillRule={0}
              propList={
                Array [
                  "fill",
                  "fillRule",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
    <Text
      ellipsizeMode="tail"
      numberOfLines={0}
      selectable={false}
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "Heavitas",
          },
          Object {
            "fontSize": 26,
          },
          Object {
            "textAlign": "left",
          },
          Object {
            "color": "#fff",
            "marginTop": 5,
          },
        ]
      }
    >
      Gegevens
    </Text>
  </View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "flex": 1,
            "padding": 25,
            "position": "relative",
            "width": 750,
          }
        }
      >
        <Text
          ellipsizeMode="tail"
          numberOfLines={0}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "Heavitas",
              },
              Object {
                "fontSize": 58,
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "marginBottom": 30,
              },
            ]
          }
        >
          Jouw eigen gegevens
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "marginBottom": 20,
            }
          }
        >
          <Text>
            ðŸ‘†ðŸ˜Œ
          </Text>
        </View>
        <Text
          ellipsizeMode="tail"
          numberOfLines={0}
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "Raleway-Regular",
                "lineHeight": 46,
              },
              Object {
                "fontSize": 29,
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "marginBottom": 30,
              },
            ]
          }
        >
          Wij van Ping Ping gebruiken de door jouw ingevulde gegevens alleen om jouw route te bepalen en de app te verbeteren. Wij zullen nooit jouw gegevens verkopen aan andere partijen.
        </Text>
        <Text
          ellipsizeMode="tail"
          numberOfLines={0}
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "Raleway-Regular",
                "lineHeight": 46,
              },
              Object {
                "fontSize": 29,
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "marginBottom": 30,
              },
            ]
          }
        >
          Wil je toch graag je inloggegevens verwijderen? Druk simpelweg op de knop hieronder om je gegevens uit ons systeem te halen. Hierdoor gaat je voortgang verloren.
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <View
            accessibilityState={
              Object {
                "disabled": false,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#f9454f",
                "borderRadius": 5,
                "flexDirection": "row",
                "justifyContent": "center",
                "opacity": 1,
                "padding": 10,
              }
            }
            testID="DELETE_DATA_BUTTON"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={20}
                bbWidth={24}
                color={4294967295}
                fill="none"
                focusable={false}
                height={20}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Object {
                      "flex": 0,
                      "height": 20,
                      "width": 24,
                    },
                  ]
                }
                tintColor={4294967295}
                vbHeight={29}
                vbWidth={24}
                width={24}
                xmlns="http://www.w3.org/2000/svg"
              >
                <RNSVGGroup
                  fill={null}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                >
                  <RNSVGPath
                    d="M22.781 2.063h-6.328l-.527-.95c-.211-.422-.686-.738-1.108-.738H8.754c-.422 0-.95.316-1.108.738l-.474.95H.844c-.475 0-.844.421-.844.843v1.688c0 .474.37.843.844.843H22.78c.422 0 .844-.369.844-.843V2.906c0-.422-.422-.844-.844-.844ZM2.795 25.002c.053 1.318 1.213 2.373 2.531 2.373h12.92c1.319 0 2.479-1.055 2.531-2.373l1.16-17.877H1.688l1.108 17.877Z"
                    fill={4294967295}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
              <Text
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "color": "#fff",
                      "fontFamily": "Heavitas",
                      "fontSize": 26,
                    },
                    Object {},
                  ]
                }
              >
                Verwijder mijn gegevens
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    statusBarTranslucent={true}
    transparent={true}
    visible={false}
  />
</View>
`;
